File: 2050-count-good-numbers/count-good-numbers.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        MOD = 10**9 + 7
        
        def power(base, exp):
            if exp == 0:
                return 1
            temp = power(base, exp // 2)
            if exp % 2 == 0:
                return (temp * temp) % MOD
            else:
                return (base * temp * temp) % MOD

        even_count = (n + 1) // 2  
        odd_count = n // 2         

        even = power(5, even_count)
        odd = power(4, odd_count)

        return (even * odd) % MOD

File: 2050-count-good-numbers/count-good-numbers.cpp
class Solution {
public:
    long MOD = 1000000007;
    long pow(long x , long n){
        if(n==0) return 1;
        long temp =pow(x,n/2);
        if(n%2==0) return (temp*temp)%MOD;
        else return (x*temp*temp)%MOD;
    }
    int countGoodNumbers(long long n) {
        long e=(n+1)/2, o=n/2;
        long f=pow(5,e)%MOD, s=pow(4,o)%MOD;
        return (int)((f*s)%MOD);
    }
};